File: src/main/java/uk/co/endofhome/javoice/Observable.java
package uk.co.endofhome.javoice;

import java.io.IOException;

public interface Observable {
    void registerObserver(Observer observer);
    void newCustomer(String name, String addressOne, String addressTwo, String postcode, String phoneNumber) throws IOException;
    void searchForCustomer(String name) throws Exception;
}

File: src/main/java/uk/co/endofhome/javoice/Observable.kt
package uk.co.endofhome.javoice

import java.io.IOException

interface Observable {
    fun registerObserver(observer: Observer)
    @Throws(IOException::class)
    fun newCustomer(name: String, addressOne: String, addressTwo: String, postcode: String, phoneNumber: String)

    @Throws(Exception::class)
    fun searchForCustomer(name: String)
}
